### PR TITLE
ISOM-1087: Setup ORM

### PR DESCRIPTION
**Pull request by harishv7**
## Problem

StarterKit already comes with Prisma. However, during the last discussion, we talked about how Prisma is limiting and the way it does some of the joins etc. is not efficient. Please point out the exact details here if you know.

Closes ISOM-1087

## Solution

As one of the solutions, we discussed the use of Prisma alongside Kysely. Kysely is a type-safe query builder (not a full-fledged ORM). Hence, for our queries we can use Kysely, and we use Prisma to define our schema and for migrations. We can use the `prisma-kysely` lib to generate the types.

This PR introduces Kysely alongside a simple example.

Note that we can still choose/opt to just go with Prisma alone.

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
  - Details ...
- [X] No - this PR is backwards compatible

## Tests

- [ ] Add a console log after the Kysely example to print out the results
- [ ] Navigate to http://localhost:3000/profile/{userId}/liked on local
- [ ] See if your terminal is printing the output

**New scripts**:

- `db:push` : this is a Prisma specific push command useful during the prototyping phase. Helps to synchronise db schema and prisma schema. Docs [here](https://www.prisma.io/docs/orm/prisma-migrate/workflows/prototyping-your-schema). Note: Not to be used in production!

**New dependencies**:

- `pg` : required to create and initialise pool for Kysely DB in `apps/studio/src/utils/database.ts`
- `kysely`: type-safe query builder
- `prisma-kysely`: helps to generate types from prisma schema

**New dev dependencies**:

- `@types/pg` : for above `pg` module

### CODE REVIEW

File: apps/studio/prisma/generated/types.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import type { ColumnType } from "kysely"
```
Comment (kishore03109) [inline apps/studio/prisma/generated/types.ts:1]: nit: use ts-morph, and generate a heading that states not to edit this file direcly as this is auto-generated. 

similar to what kr did (here)[https://github.com/opengovsg/activesg/pull/892/files]

File: apps/studio/prisma/generated/types.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  T extends ColumnType<infer S, infer I, infer U>
+    ? ColumnType<S, I | undefined, U>
+    : ColumnType<T, T | undefined, T>
+export type Timestamp = ColumnType<Date, Date | string, Date | string>
+
+export type Accounts = {
+  id: string
+  provider: string
+  providerAccountId: string
+  user_id: string
```
Comment (kishore03109) [inline apps/studio/prisma/generated/types.ts:12]: hmm this means that we will end up using snake case throughout our codebase. Is there a setting we can use to make camelCase instead? (iirc, that is the default, not sure why this is snek case)

File: apps/studio/prisma/seed.ts [hunk 3]
```diff
@@ -3,19 +3,13 @@
...
-    console.error(e)
-    process.exit(1)
-  })
-  .finally(async () => {
-    await prisma.$disconnect()
-  })
+main().catch((e) => {
+  console.error(e)
+  process.exit(1)
+})
```
Comment (kishore03109) [inline apps/studio/prisma/seed.ts:15]: shall we .finally(disconnect)? if not failed instances might have dangling connections liao

File: apps/studio/prisma/generated/types.ts (hunk 2 above)
Comment (harishv7) [reply apps/studio/prisma/generated/types.ts:12]: hmm this represents our DB right, which is in snake_case?

File: apps/studio/prisma/seed.ts (hunk 3 above)
Comment (harishv7) [reply apps/studio/prisma/seed.ts:15]: oh yes, will add

File: apps/studio/prisma/generated/types.ts (hunk 2 above)
Comment (kishore03109) [reply apps/studio/prisma/generated/types.ts:12]: https://github.com/valtyr/prisma-kysely
think we just need to enable a flag for the camel case plugin bah

File: apps/studio/prisma/generated/types.ts (hunk 1 above)
Comment (seaerchin) [reply apps/studio/prisma/generated/types.ts:1]: I think the point of `ts-morph` was to generate typings automatically cos they were using postgis which wasn't supported.

We can still do it, but given that this type is already automatically generated (and lives in a `generated` folder), there might not be much incremental value

File: apps/studio/prisma/generated/types.ts (hunk 2 above)
Comment (harishv7) [reply apps/studio/prisma/generated/types.ts:12]: i removed the `@map` on the schema level, so all is in camelCamse now

File: apps/studio/prisma/seed.ts (hunk 3 above)
Comment (harishv7) [reply apps/studio/prisma/seed.ts]: done

File: apps/studio/prisma/generated/types.ts (hunk 1 above)
Comment (harishv7) [reply apps/studio/prisma/generated/types.ts:1]: yea, skipping this for now

File: apps/studio/.gitignore [hunk 4]
```diff
@@ -41,7 +41,7 @@ yarn-error.log*
 playwright/test-results
 
 # env
-.env
+.env*
```
Comment (kishore03109) [inline apps/studio/.gitignore:44]: what about .env.example? we using env-example ah?

**Review by kishore03109 — APPROVED** [review]

File: apps/studio/.gitignore (hunk 4 above)
Comment (harishv7) [reply apps/studio/.gitignore]: will make it less restrictive first

**Pull request merged**